src/Simplify.cpp

```diff
@@ -729,7 +741,19 @@ class Simplify : public IRMutator {
         } else if (broadcast_a && broadcast_b) {
             // Broadcast + Broadcast
             expr = Broadcast::make(mutate(broadcast_a->value - broadcast_b->value),
-                                 broadcast_a->lanes);
+                                   broadcast_a->lanes);
+        } else if (cast_a && cast_b &&
+                   cast_a->value.type() == cast_b->value.type() &&
+                   cast_a->type.is_int() &&
+                   cast_b->type.is_int() &&
+                   cast_a->value.type().is_int() &&
+                   cast_b->value.type().is_int() &&
+                   no_overflow(cast_a->type) &&
+                   no_overflow(cast_b->type) &&
+                   no_overflow(cast_a->value.type()) &&
+                   no_overflow(cast_b->value.type())) {
+            // cast(t, a) - cast(t, b) -> cast(t, a - b)
```

Comment:
The particular place we need this is for the correctness_interleave test, `interleave.cpp:271`. Specifically, in `Deinterleave.cpp`, the `Interleaver` visitor for `Block`s has this check:

        Expr diff = simplify(ri->base - r0->base);
        const int64_t *offs = as_const_int(diff);

Which will fail for the `unrolled` test without the simplification rule we're talking about. Here is how it evaluates with the new rule:

    ((int64((unrolled.s0.s1.base + 1)) + (int64(unrolled.s0.y.yi.base)*int64(unrolled.stride.1))) - (int64(unrolled.min.0) + (int64(unrolled.min.1)*int64(unrolled.stride.1)))) - ((int64(unrolled.s0.s1.base) + (int64(unrolled.s0.y.yi.base)*int64(unrolled.stride.1))) - (int64(unrolled.min.0) + (int64(unrolled.min.1)*int64(unrolled.stride.1)))) = 1

and without it, this fails to resolve to a constant.